core/src/com/geekbrains/td/MonsterEmitter.java

```diff
@@ -1,12 +1,47 @@
 package com.geekbrains.td;
 
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.math.MathUtils;
+
+import java.util.LinkedList;
 
 public class MonsterEmitter extends ObjectPool<Monster> {
     private GameScreen gameScreen;
+    private float monsterTimer;
+    private LinkedList<MonsterWaveType> waveList;
+    private MonsterWaveType currentWave;
+    private int emittedMonsters;
+    private Monster singleMonster;
+    private boolean isSingleMonsterStatus;
+    private boolean isSingleFirstTime;
 
     public MonsterEmitter(GameScreen gameScreen) {
+        isSingleFirstTime = true;
+        isSingleMonsterStatus = false;
         this.gameScreen = gameScreen;
+        waveList = new LinkedList<MonsterWaveType>();
+        resetGeneration();
+    }
+
+    public boolean getSingleMonsterStatus() {
+        return isSingleMonsterStatus;
+    }
+
+    public void setSingleMonster(Monster singleMonster) {
+        this.singleMonster = singleMonster;
+    }
+
+    public LinkedList<MonsterWaveType> getWaveList() {
+        return waveList;
+    }
+
+    private void resetGeneration() {
+        monsterTimer = 0;
+        emittedMonsters = 0;
+
+        currentWave = null;
```

Comment:
Using `null` is generally ok, but I would recommend considering `java.util.Optional`. Using Optional instead of null is like good manners that others will definitely pay attention to.